### PR TITLE
Merging master into XML converter

### PR DESCRIPTION
**Pull request by AsherGlick**
Resolved the merge conflicts between xml converter and master so that xml converter can more easily be merged back into master when it is completed.

When merged, this PR should be fastforwarded, no need to create a merge commit for a merge commit.

### CODE REVIEW

File: Spatial.gd [hunk 1]
```diff
@@ -255,6 +321,10 @@ func decode_context_packet(spb: StreamPeerBuffer):
 	# this to just be a radian to degree conversion.
 
 	if self.map_id != old_map_id:
+		print("New Map")
+
+		print("Saving Old Map")
+		self.markerdata[str(old_map_id)] = data_from_renderview()
```
Comment (klingbolt) [inline Spatial.gd:327]: This line will crash. Markerdata is now defined on line 365 as ```var markerdata = Waypoint.Waypoint.new()```

File: Spatial.tscn [hunk 2]
```diff
@@ -600,125 +601,158 @@ disabled = true
 text = "Reverse Path Direction"
 
 [node name="SettingsDialog" type="WindowDialog" parent="Control/Dialogs"]
+visible = true
```
Comment (klingbolt) [inline Spatial.tscn:604]: This should be false. Godot gives a node configuration warning that pop ups will be hidden on run time

Comment (klingbolt) [inline Spatial.tscn:604]:
```suggestion
visible = false
```

File: Spatial.gd (hunk 1 above)
Comment (klingbolt) [inline Spatial.gd:327]:
```suggestion
```

**Review by klingbolt — CHANGES_REQUESTED** [review]

File: Spatial.tscn (hunk 2 above)
Comment (AsherGlick) [reply Spatial.tscn]: accepting suggestion to remove the line, it will be re-added when saving is recreated.

**Review by klingbolt — APPROVED** [review]

**Comment by AsherGlick** [comment]
Woops, it looks like I made a merge commit. Oh well, not going to fix it now we will just live.

**Pull request merged**